File: tai-orm/src/main/kotlin/tai/orm/upsert/builder/UpsertFunctionBuilder.java
package tai.orm.upsert.builder;

import tai.orm.core.BuilderContext;
import tai.orm.upsert.UpsertFunction;

/**
 * Created by Jango on 2017-01-21.
 */
@FunctionalInterface
public interface UpsertFunctionBuilder {
    UpsertFunction build(String entity, BuilderContext<UpsertFunction> functionMap);
}

File: tai-orm/src/main/kotlin/tai/orm/upsert/builder/UpsertFunctionBuilder.kt
package tai.orm.upsert.builder

import tai.orm.core.BuilderContext
import tai.orm.upsert.UpsertFunction

/**
 * Created by Jango on 2017-01-21.
 */
@FunctionalInterface
interface UpsertFunctionBuilder {
    fun build(entity: String, functionMap: BuilderContext<UpsertFunction>): UpsertFunction
}